deepchem/models/torch_models/ferminet.py

```diff
@@ -26,48 +27,126 @@ def test_f(x: np.ndarray) -> np.ndarray:
+        n_two: List
+            List of hidden units for the two-electron stream in each layer
+        determinant: int
+            Number of determinants for the final solution
         """
-    Parameters:
-    -----------
-    nucleon_pos: torch.Tensor
-        Torch tensor containing nucleus information of the molecule
-    nuclear_charge: torch.Tensor
-        Torch tensor containing the number of electron for each atom in the molecule
-    spin: tuple
-        Tuple in the format of (up_spin, down_spin)
-    inter_atom: torch.Tensor
-        Torch tensor containing the pairwise distances between the atoms in the molecule
-    n_one: List
-      List of hidden units for the one-electron stream in each layer
-    n_two: List
-      List of hidden units for the two-electron stream in each layer
-    determinant: int
-      Number of determinants for the final solution
-    """
         super(Ferminet, self).__init__()
         if len(n_one) != len(n_two):
             raise ValueError(
-                "The number of layers in one-electron and two-electron stream should be equal"
-            )
+              "The number of layers in one-electron and two-electron stream should be equal"
+        )
         else:
             self.layers = len(n_one)
+
         self.nucleon_pos = nucleon_pos
         self.determinant = determinant
+        self.batch_size = batch_size
         self.spin = spin
-        self.inter_atom = inter_atom
+        self.total_electron = spin[0] + spin[1]
+        self.nuclear_charge = nuclear_charge
+        self.v = nn.ModuleList()
+        self.w = nn.ModuleList()
+        self.envelope_w = nn.ParameterList()
+        self.envelope_g = nn.ParameterList()
+        self.sigma = nn.ParameterList()
+        self.pi = nn.ParameterList()
         self.n_one = n_one
         self.n_two = n_two
-        self.determinant = determinant
+
+        self.v.append(nn.Linear(8+3*4*self.nucleon_pos.size()[0],n_one[0],bias=True))
+        self.w.append(nn.Linear(4,n_two[0],bias=True))
+        for i in range(1,self.layers):
+            self.v.append(nn.Linear(3*n_one[i-1]+2*n_two[i-1],n_one[i],bias=True))
+            self.w.append(nn.Linear(n_two[i-1],n_two[i],bias=True))
+
+        for i in range(self.determinant):
+            for j in range(self.total_electron):
+                self.envelope_w.append(torch.nn.init.kaiming_uniform(torch.empty(n_one[-1],1)).squeeze(-1))
+                self.envelope_g.append(torch.nn.init.uniform(torch.empty(1)).squeeze(0))
+                for k in range(self.nucleon_pos.size()[0]):
+                    self.sigma.append(torch.nn.init.uniform(torch.empty(self.nucleon_pos.size()[0],1)).squeeze(0))
+                    self.pi.append(torch.nn.init.uniform(torch.empty(self.nucleon_pos.size()[0],1)).squeeze(0))
+
+    def forward(self, input):
+        # creating one and two electron features
+        self.input = torch.from_numpy(input)
```

Comment:
This is quite complicated. Can we split up into smaller helper functions that have simpler descriptions?